TypedSyntax/src/node.jl

```diff
@@ -306,15 +307,22 @@ end
 
 function getsrc(@nospecialize(f), @nospecialize(t))
     srcrts = code_typed(f, t; debuginfo=:source, optimize=false)
-    return only(srcrts)
+    src, rt = only(srcrts)
+    if hasfield(typeof(src), :parent)
+        mi = src.parent
+    else
+        mi = Base.method_instance(f, t)
+    end
```

Comment:
I'm a bit uncertain about using a separate `method_instance` lookup than the one implicit in `code_typed`. That sounds like an opportunity for desynchronization.

However, I don't think we have a MI-based code_typed entry point yet so I don't believe there's another alternative.

Reply:
> However, I don't think we have a MI-based code_typed entry point yet

I think it would be reasonable to add one. I agree, it would be better to look up the method instance first and use that to query the source.

Reply:
Implemented the refactoring at 707a1818d16ac3a2d9cecd7fc916b4ca70fada7b.